src/containerapp/azext_containerapp/_utils.py

```
@@ -531,33 +531,40 @@ def _generate_log_analytics_if_not_provided(cmd, logs_customer_id, logs_key, loc
     if logs_customer_id is None and logs_key is None:
         logger.warning("No Log Analytics workspace provided.")
         _validate_subscription_registered(cmd, LOG_ANALYTICS_RP)
+
         try:
             log_analytics_client = log_analytics_client_factory(cmd.cli_ctx)
             log_analytics_shared_key_client = log_analytics_shared_key_client_factory(cmd.cli_ctx)
+        except Exception as ex:
+            handle_raw_exception(ex)
 
-            log_analytics_location = location
-            try:
-                _ensure_location_allowed(cmd, log_analytics_location, LOG_ANALYTICS_RP, "workspaces")
-            except Exception:  # pylint: disable=broad-except
-                log_analytics_location = _get_default_log_analytics_location(cmd)
+        log_analytics_location = location
+        try:
+            _ensure_location_allowed(cmd, log_analytics_location, LOG_ANALYTICS_RP, "workspaces")
+        except Exception:  # pylint: disable=broad-except
+            log_analytics_location = _get_default_log_analytics_location(cmd)
```

I know this isn't too related to your change, but we should definitely only be excepting `ValidationError` here since that's what `_ensure_location_allowed` raises 

We should also revisit `_ensure_location_allowed`. I took a look at the implementation and the whole function basically is under a try/except and it ignores all exceptions besides ValidationError -- not sure why that is. Seems like we shouldn't be burying errors like that. You can just put a TODO comment on it for now if you don't have time to fix it and retest (and it's sort of out of the scope of this change)